lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -572,7 +582,7 @@ def modify_db_instance(module, conn):
 
     # guess that this changed the DB, need a way to check
     diff = existing_instance.diff(instance.data)
-    changed = not not diff
+    changed = not diff
```

Comment:
`diff == None` => `not diff == True` => `changed = True`

which is wrong. If diff is empty, changed should be False (using `not not` to convert it to a boolean)